channel.go

```diff
@@ -94,6 +94,12 @@ type ChannelOptions struct {
 	// This is an unstable API - breaking changes are likely.
 	RelayTimerVerification bool
 
+	// EnableMPTCP enables MPTCP for TCP network connection to increase reliability.
+	// It requires underlying operating system support MPTCP.
+	// If EnableMPTCP is false or no MPTCP support, the connection will use normal TCP.
+	// It's set to false by default.
```

Comment:
Let's also call out in the comment passing your own `Dialer` means this value will be ignore on the outbound side. That is what the implementation says down below.
```
// If both a Dialer is passed as option, this value will be ignored.
```

Reply:
given @jronak 's comments about how we pass in listeners/dialers into tchannel explicitly for yarpc, why not just do this consistently rather than adding option here? I don't get why we would add this while not having similar capabilities for other connection-level configuration.

Reply:
for services that are not on yarpc, they are calling `ListenAndServe()` and we should have this parity.